src/providers/details/tabs/Hosts/components/HostsNetworksSetPassword.tsx

```diff
@@ -0,0 +1,63 @@
+import { type Dispatch, type FC, type FormEvent, type SetStateAction, useState } from 'react';
+import { validateNoSpaces } from 'src/modules/Providers/utils/validators/common';
+
+import { FormGroupWithHelpText } from '@components/common/FormGroupWithHelpText/FormGroupWithHelpText';
+import { Button, ButtonVariant, InputGroup, TextInput } from '@patternfly/react-core';
+import { EyeIcon, EyeSlashIcon } from '@patternfly/react-icons';
+import { useForkliftTranslation } from '@utils/i18n';
+
+import type { ValidationState } from './utils/types';
+
+type HostsNetworksSetPasswordProps = {
+  password: string;
+  setPassword: Dispatch<SetStateAction<string>>;
+};
+
+const HostsNetworksSetPassword: FC<HostsNetworksSetPasswordProps> = ({ password, setPassword }) => {
+  const { t } = useForkliftTranslation();
+
+  const [passwordValidation, setPasswordValidation] = useState<ValidationState>('default');
```

Comment:
Is ValidateState an enum? Or, please convert to an enum and use it as an initial value instead of hardcoding it